File: 4258-construct-uniform-parity-array-ii/construct-uniform-parity-array-ii.py
class Solution(object):
    def uniformArray(self, nums1):
        """
        :type nums1: List[int]
        :rtype: bool
        """
        odd=[]
        for x in nums1:
            if(x%2==1):
                odd.append(x)
        if(len(odd)==0):
            return True
        smallest_odd=min(odd)
        for x in nums1:
            if (x%2==0 and x<smallest_odd):
                return False
        return True

        

File: 4258-construct-uniform-parity-array-ii/construct-uniform-parity-array-ii.java
class Solution {
    public boolean uniformArray(int[] nums1) {

        // Step 1: Find the minimum element
        int min_val = nums1[0];

        for (int x : nums1) {
            if (x < min_val) {
                min_val = x;
            }
        }

        // Step 2: If minimum is odd
        if (min_val % 2 != 0) {
            return true;
        }

        // Step 3: If minimum is even,
        // all elements must be even
        for (int x : nums1) {
            if (x % 2 != 0) {
                return false;
            }
        }

        return true;
    }
}